skytemple_files/_resources/patches/asm_patches/happylappy_asm_mods/ai_item_optimizations/common/optimizations_ov29.asm

```diff
@@ -0,0 +1,402 @@
+    ldrb r4,[r6,#0xbf]; Get BurnClass Status (which contains all poison types)   
+    cmp r4,#0x2; If Poisoned
+    cmpne r4,#0x3; Else If Badly Poisoned
+    bne odds_0; Only bother using if Poisoned!  
+    b Adjacent_100_50;
+
+IsBlinkerSeed:
+    mov r0,r7    
+    mov r1,#0x1    
+    bl IsBlinded
+    cmp r0,#0x1    
+    b False_Adjacent_80_5;
+
+IsWarpSeed:
+    bl Adjacent2Enemy;
+    beq odds_40;    
+    b odds_5;   
+
+IsSleepSeed:
+    mov r0,r7;
+    bl IsMonsterSleeping
+    b False_Adjacent_80_5;
+
+IsChestoBerry:
+    ldrb r4,[r6,#0xbd]; Get SleepClass Statuses   
+    cmp r4,#0x2; Is Sleepless?
+    beq odds_0; Don't try it!    
+    b odds_5; Otherwise, 5% at all times (???)
+
+IsJoySeed: 
+    ldrb r4,[r6,#0xa]; Get current level   
+    cmp r4,#0x63; Compare against 99...
+    bgt odds_0;
+IsGinseng:
+    b odds_80;
+
+IsRawstBerry:
+    ldrb r4,[r6,#0xbf]; Get BurnClass Statuses   
+    cmp r4,#0x1; If Burned...
+    beq odds_50;
+    b odds_0;
+
+IsDoomSeed:
+    ldrb r4,[r6,#0xa]; Get current level   
+    cmp r4,#0x1; If level greater than 1...
+    bgt odds_80;
+    b odds_0;
+
+IsStunSeed:
+    ldrb r4,[r6,#0xc4]; Get FreezeClass Statuses  
+    cmp r4,#0x6; If Stunned...
+    b False_Adjacent_80_5;
+
+IsDoughSeed:
+    ldrb r4,[r6,#0x6]; Check if on the team   
+    cmp r4,#0x0; If on team...
+    bne odds_0; Enemies will never use this!!!
+    b odds_100;
+   
+IsSlipSeed:
+    ldrsh r0,[r6,#0x2]; Get true species (Ditto)  
+    ldrb r1,[r6,0xef]; Get Invisibility Class Status (Slip is 0x4, 0x5 and beyond is unused!)
+    lsr r1,#0x2; Is Slipping?
+    bl GetMobilityTypeCheckSlip
+    cmp r0,#0x0
+    cmpne r0,#0x4
+    b False_Adjacent_10_100;
+
+IsGonePebble: 
+    ldrb r4,[r6,#0xd5]; Get ReflectClass Statuses   
+    cmp r4,#0x9; If Enduring...
+    beq odds_0;    
+Adjacent_0_80:
+    bl Adjacent2Enemy;
+    b True_80_False_0;
+
+IsViolentSeed:
+    ldrsh r4,[r6,#0x26]; Retrieve SPA value (Bug? Should also check ATK)   
+    cmp r4,#0x14; 
+    bge odds_0;    
+    b odds_80;
+  
+IsVanishSeed:
+    ldrb r4,[r6,#0xef]; Get InvisibleClass Statuses   
+    cmp r4,#0x1; If Invisible...
+    bne odds_0;
+    b odds_80;
+  
+IsApple:
+    bl ComputeHunger;
+    cmp r4,#0xa; If belly below 10...
+    blt odds_100;
+    b odds_0;  
+
+IsDropEye:
+    ldrb r4,[r6,#0xf1]; Get BlindClass Status 
+    cmp r4,#0x4; If DropEyed...
+    beq odds_0;
+    b odds_80;
+
+IsGaggleOrYRaySpecs:
+    cmp r5,#0x1; If throwing...
+    b True_80_False_0;
+
+IsGravelyrock:
+    ldrsh r4,[r6,#0x4]; Get Apparant Species, This is bugged to only check males!
+    cmp r4,#0x1e0; Is Bonsly?    
+    cmpne r4,#0xb9; Is Sudowoodo?
+    bne odds_0;   
+    cmp r5,#0x1; If Throwing...   
+    beq odds_70;
+    ; Else continue to odds_100
+odds_100:
+    add r0,#20;
+odds_80:
+    add r0,#10;
+odds_70:
+    add r0,#20;
+odds_50:
+    add r0,#10;
+odds_40:
+    add r0,#10;
+odds_30:
+    add r0,#20;
+odds_10:
+    add r0,#5;
+odds_5:
+    add r0,#5;
```

Comment:
This is quite confusing to follow. Why not replace `b odds_xxx` with `mov r0,xxx; b exit`? Do you need the extra few instructions that badly?
If so, I'd at least add a comment at the start of this section explaining what it does.